provider/__init__.py

```diff
@@ -1 +1 @@
-__version__ = '0.5.0'
+__version__ = '0.6.0'
```

Comment:
This is the first I have heard of Item 4. Considering we have no clear roadmap for when we will reach 1.x (for this, and pretty much all of our other libraries), I would much rather err on the side of caution and make this a major version bump.